libs/core/algorithms/tests/regressions/reduce_6647.cpp

```diff
@@ -0,0 +1,76 @@
+// proxy_reduce_test.cpp
+#include <hpx/hpx_main.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/algorithms/reduce.hpp>
+#include <vector>
+
+// A custom proxy type for testing:
+// Its implicit conversion to int adds an extra offset (+1)
+struct Proxy
+{
+    int value;
+    Proxy(int v)
+      : value(v)
+    {
+    }
+    Proxy()
+      : value(0)
+    {
+    }
+
+    // Implicit conversion: note the +1 here.
+    operator int() const
+    {
+        return value + 1;
+    }
+};
+
+// Define addition for Proxy objects so that when added as Proxy the raw values are summed.
+Proxy operator+(const Proxy& a, const Proxy& b)
+{
+    return Proxy(a.value + b.value);
+}
+
+// Also, allow addition with int to combine the Proxy with an int.
+int operator+(int a, const Proxy& b)
+{
+    return a + b.value;
+}
+
+int operator+(const Proxy& b, int a)
+{
+    return b.value + a;
+}
```

Comment:
Are these operators necessary? Your proxy object has an implicit conversion operator to `int` already.